### PR TITLE
[READY] Adds monkey sentience caps!

### PR DESCRIPTION
**Pull request by tralezab**
### inspired by this thread https://tgstation13.org/phpBB/viewtopic.php?f=9&t=26834

Cargo can buy a science crate with two monkey helmets. for humans, they just hurt to put on but otherwise act as normal helmets. for monkeys, they drill into the brain and wipe out any sentience before adding a new one. Taking off the helmet from this point removes the sentience. Yes, you can use this to punish a monkey with mindwipe.

The helmets themselves are one use to prevent candidate spam, but posibrains are better for "spamming" candidates (you can opt out) anyways so i'm not worried about that

polish:
- [x] add more messages for failed pull
- [x] sprites
- [x] sounds (beeps n buzzes)
- [x] fix some stuff about quickly equipping and unequipping breaking shit
- [x] fix the edge cases athath pointed out
- [x] remove it from working on smart monkeys

## Why It's Good For The Game

There are so many good ideas I have for these already, cargo militarizing the monkeys, scientists using monkeys as fodder to fix things on the station, captain getting dethroned by sentiented-rev-monkeys, oh my god

## Changelog
:cl:
add: Cargo can purchase monkey sentience helmets under the science tab!
/:cl:

### CODE REVIEW

**Comment by Doosteroni** [comment]
What if you could like...email it and then the monkey is like...slaved or like if a human wears it, it drills their brain and kills them.

**Comment by Tlaltecuhtli** [comment]
what happens if applied to a ling monky

**Comment by tralezab** [comment]
@Tlaltecuhtli ling monkey gets a new player

**Comment by tmtmtl30** [comment]
is there admin / antag policy hammered out for this

**Comment by Tlaltecuhtli** [comment]
policy is that if you are a monkey you are valid

**Comment by necromanceranne** [comment]
Not entirely sure this should work on monkeys with a mind at all. Because I can already see an established meta of monkeying, slapping them with this and then unmonkeying.

I'm assuming it removes antag right?

**Comment by Ratvar** [comment]
I would guess that if you unmonkey such a monkey, it loses sentience and you get an empty body, preventing greytider, metafriend and conversion antag abuse to an extent?

**Comment by SkeletalElite** [comment]
> @Tlaltecuhtli ling monkey gets a new player

It shouldn't do anything to ling monkeys because vestigial brain (even though its not working right now)

**Comment by tralezab** [comment]
@SkeletalElite On god this code works

**Comment by SkeletalElite** [comment]
> On god this code works

I was talking about vestigial brains for lings not working

**Comment by tralezab** [comment]
oh yeah!

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -335,6 +335,87 @@
 	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH | PEPPERPROOF
 	strip_delay = 80
 
+//monkey sentience caps
+
+/obj/item/clothing/head/helmet/monkey_sentience
+	name = "monkey mind magnification helmet"
+	desc = "A fragile, circuitry embedded helmet to boost the intelligence on monkeys to higher levels. You see several warning labels..."
```
Comment (ATH1909) [inline code/modules/clothing/head/helmet.dm:342]:
```suggestion
	desc = "A fragile, circuitry embedded helmet for boosting the intelligence of a monkey to a higher level. You see several warning labels..."

```

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -335,6 +335,87 @@
...
+	var/list/candidates = pollCandidatesForMob("Do you want to play as a mind magnified monkey?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, magnification, POLL_IGNORE_SENTIENCE_POTION) - removed_candidate
+	polling = FALSE
+	if(!candidates.len)
+		magnification = null
+		visible_message("<span class='notice'>[src] falls silent. Maybe you should try again later?</span>")
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+	var/mob/picked = pick(candidates)
+	magnification.key = picked.key
+	playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
+	to_chat(magnification, "<span class='notice'>You're a mind magnified monkey! Protect your helmet with your life, if you lose it your sentience goes with it.</span>")
```
Comment (ATH1909) [inline code/modules/clothing/head/helmet.dm:385]:
```suggestion
	to_chat(magnification, "<span class='notice'>You're a mind magnified monkey! Protect your helmet with your life- if you lose it, your sentience goes with it!</span>")

```

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -335,6 +335,87 @@
...
+		return
+	if(!polling)//put on a viable head, but taken off after polling finished.
+		if(magnification.ckey)
+			to_chat(magnification, "<span class='userdanger'>You feel your flicker of sentience ripped away from you, as everything becomes dim...</span>")
+			magnification.ghostize(FALSE)
+		if(prob(10))
+			switch(rand(1,4))
+				if(1) //blood rage
+					magnification.aggressive = TRUE
+				if(2) //brain death (death)
```
Comment (ATH1909) [inline code/modules/clothing/head/helmet.dm:403]: make this kill them via brain damage

**Comment by ATH1909** [comment]
me and the boys turning the wizard into a monkey, then replacing his ckey with someone else's ckey so that someone else can steal his wizard round (and still know/cast the wizard's spells, from what I can tell)

**Comment by ATH1909** [comment]
should it be "mind magnification" or "mind-magnification"?

**Comment by ATH1909** [comment]
also, while you're at it, could you fix this bug (in either this PR or another one), since it's also related to uplifting monkeys/monkeymen?

https://github.com/tgstation/tgstation/issues/49203

**Comment by ATH1909** [comment]
also also, what happens if a mind-magnified monkey gets humanized by, say, mutadone?

**Comment by Shadowflame909** [comment]
Happy Happy Joy Joy helmets from ren and stimpy confirmed!

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -335,6 +335,87 @@
...
+		return
+	if(!ismonkey(user))
+		var/mob/living/something = user
+		to_chat(something, "<span class='boldnotice'>You feel a stabbing pain in the back of your head for a moment.</span>")
+		something.apply_damage(5,BRUTE,BODY_ZONE_HEAD,FALSE,FALSE,FALSE) //notably: no damage resist (it's in your helmet), no damage spread (it's in your helmet)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+		return
+	magnification = user //this polls ghosts
+	var/removed_candidate = magnification.ckey //don't poll the guy you're about to ghost
+	if(removed_candidate)
```
Comment (ATH1909) [inline code/modules/clothing/head/helmet.dm:370]: I thought you said you removed this?

**Comment by trollbreeder** [comment]
image

File: code/modules/clothing/head/helmet.dm (hunk 4 above)
Comment (tralezab) [reply code/modules/clothing/head/helmet.dm:370]: i'm gunna gosh

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -335,6 +335,89 @@
...
+
+/obj/item/clothing/head/helmet/monkey_sentience/Destroy()
+	. = ..()
+	disconnect()
+
+/obj/item/clothing/head/helmet/monkey_sentience/proc/disconnect()
+	if(!magnification) //not put on a viable head
+		return
+	if(!polling)//put on a viable head, but taken off after polling finished.
+		if(magnification.ckey)
```
Comment (nemvar) [inline code/modules/clothing/head/helmet.dm:398]: Look for client.

File: code/modules/clothing/head/helmet.dm [hunk 6]
```diff
@@ -335,6 +335,89 @@
...
+				if(2) //brain death (death)
+					magnification.death()
+				if(3) //primal gene (gorilla)
+					magnification.gorillize()
+				if(4) //genetic mass susceptibility (gib)
+					magnification.gib()
+	//either used up correctly or taken off before polling finished (punish this by destroying the helmet)
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+	playsound(src, "sparks", 100, TRUE)
+	visible_message("<span class='warning'>[src] fizzles and breaks apart!</span>")
```
Comment (nemvar) [inline code/modules/clothing/head/helmet.dm:414]: ``magnification = null``

File: code/modules/clothing/head/helmet.dm [hunk 7]
```diff
@@ -335,6 +335,89 @@
...
+		return
+	magnification = user //this polls ghosts
+	var/removed_candidate = magnification.ckey //don't poll the guy you're about to ghost
+	if(removed_candidate)
+		to_chat(magnification, "<span class='userdanger'>You feel a stabbing pain in the back of your head for a moment as [src] wipes your personality to a blank slate.</span>")
+		magnification.ghostize(FALSE)
+	visible_message("<span class='warning'>[src] powers up!</span>")
+	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+	polling = TRUE
+	var/list/candidates = pollCandidatesForMob("Do you want to play as a mind magnified monkey?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, magnification, POLL_IGNORE_SENTIENCE_POTION) - removed_candidate
```
Comment (nemvar) [inline code/modules/clothing/head/helmet.dm:377]: ``pollCandidatesForMob`` returns a list of observers, removing a ckey from it doesn't work.
Instead remove the return of ghostize.

File: code/modules/clothing/head/helmet.dm [hunk 8]
```diff
@@ -335,6 +335,89 @@
...
+	//either used up correctly or taken off before polling finished (punish this by destroying the helmet)
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+	playsound(src, "sparks", 100, TRUE)
+	visible_message("<span class='warning'>[src] fizzles and breaks apart!</span>")
+	magnification = null
+	new /obj/effect/decal/cleanable/ash/crematorium(src.loc) //just in case they're in a locker or other containers it needs to use crematorium ash, see the path itself for an explanation
+
+/obj/item/clothing/head/helmet/monkey_sentience/dropped(mob/user)
+	. = ..()
+	qdel(src)//runs disconnect code
```
Comment (ATH1909) [inline code/modules/clothing/head/helmet.dm:419]: so if you take it out of the crate but then put it on a table, on the floor, in your backpack, etc., it just disintegrates on the spot?

Comment (tralezab) [reply code/modules/clothing/head/helmet.dm:419]: 🤤 oh yea

**Comment by tralezab** [comment]
these conflicts are killing me man arrrrrrr

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
https://github.com/tgstation/tgstation/blob/master/.github/POLICYCONFIG.md

**Comment by tralezab** [comment]
aaaaaaaaah its doooooone

**Comment by Angustmeta** [comment]
Is the PR body still accurate?

**Pull request merged**